cores/esp8266/core_esp8266_phy.c

```diff
@@ -88,7 +88,11 @@ static const uint8_t ICACHE_FLASH_ATTR phy_init_data[128] =
     // 0: 40MHz
     // 1: 26MHz
     // 2: 24MHz
+    #ifdef F_CRYSTAL
```

Comment:
Let me suggest an approach which will not require f_crystal value to be added to all the boards... Since 26MHz is pretty much the default for the ESP8266 based designs, we only need to add `f_crystal=40` to the boards which use 40 MHz crystal.

Then you can have something along these lines:
```c++
#if F_CRYSTAL == 40000000
    [48] = 0,
#else
    [48] = 1,
#endif
```